corehq/apps/case_importer/templates/case_importer/partials/ko_import_history.html

```diff
@@ -18,14 +18,18 @@
     <!--ko foreach: case_uploads-->
     <tr>
         <td class="col-md-1">
-            <!--ko if: task_status().state == $root.states.SUCCESS-->
+            <!--ko if: task_status().state == $root.states.SUCCESS && task_status().result-->
                 <!--ko if: _.isEmpty(task_status().result.errors)-->
                 <span class="label label-success">{% trans "Success" %}</span>
                 <!--/ko-->
                 <!--ko if: !_.isEmpty(task_status().result.errors)-->
                 <span class="label label-warning">{% trans "Success with warnings" %}</span>
                 <!--/ko-->
             <!--/ko-->
+            {# The following failed state is necessary to handle legacy errors #}
+            <!--ko if: task_status().state == $root.states.SUCCESS && !task_status().result-->
+            <span class="label label-danger">{% trans "Failed" %}</span>
+            <!--/ko-->
```

Comment:
This duplicates the span below. However I thought this was cleaner than creating a larger, more complicated if statemnet (knockouts lack of else statements feels a little weird to me)